cypher-shell/src/main/java/org/neo4j/shell/Main.java

```diff
@@ -75,37 +77,57 @@ void startShell(@Nonnull CliArgs cliArgs) {
+       System.exit( runShell( cliArgs ) );
+    }
+
+    int runShell(@Nonnull CliArgs cliArgs)
+    {
+        Logger logger = new AnsiLogger( cliArgs.getDebugMode() );
+        PrettyConfig prettyConfig = new PrettyConfig( cliArgs );
         ConnectionConfig connectionConfig = new ConnectionConfig(
                 cliArgs.getScheme(),
                 cliArgs.getHost(),
                 cliArgs.getPort(),
                 cliArgs.getUsername(),
                 cliArgs.getPassword(),
                 cliArgs.getEncryption(),
-                cliArgs.getDatabase());
+                cliArgs.getDatabase() );
+        try
+        {
+            CypherShell shell = new CypherShell( logger, prettyConfig, ShellRunner.shouldBeInteractive( cliArgs ),
+                    cliArgs.getParameters() );
+
+            //if user is attempting a password change via:
+            // cypher-shell -d system -u user -p pass 'ALTER CURRENT USER SET PASSWORD from "pass" to "newPass"  ';
+            //we should allow this command to run without being prompted for a new password
+            if (isAttemptingPasswordChange( cliArgs ) ) {
+                shell.runSingleCommand( connectionConfig, cliArgs.getCypher().orElse( "" ) );
+                return EXIT_SUCCESS;
+            }
 
-        try {
-            CypherShell shell = new CypherShell(logger, prettyConfig, ShellRunner.shouldBeInteractive( cliArgs ), cliArgs.getParameters());
             // Can only prompt for password if input has not been redirected
-            connectMaybeInteractively(shell, connectionConfig, isInputInteractive(), isOutputInteractive());
+            connectMaybeInteractively( shell, connectionConfig, isInputInteractive(), isOutputInteractive() );
 
             // Construct shellrunner after connecting, due to interrupt handling
-            ShellRunner shellRunner = ShellRunner.getShellRunner(cliArgs, shell, logger, connectionConfig);
+            ShellRunner shellRunner = ShellRunner.getShellRunner( cliArgs, shell, logger, connectionConfig );
 
-            CommandHelper commandHelper = new CommandHelper(logger, shellRunner.getHistorian(), shell);
+            CommandHelper commandHelper = new CommandHelper( logger, shellRunner.getHistorian(), shell );
 
-            shell.setCommandHelper(commandHelper);
+            shell.setCommandHelper( commandHelper );
 
-            int code = shellRunner.runUntilEnd();
-            System.exit(code);
-        } catch (Throwable e) {
-            logger.printError(e);
-            System.exit(1);
+            return shellRunner.runUntilEnd();
+        }
+        catch ( Throwable e )
+        {
+            logger.printError( e );
+            return EXIT_FAILURE;
         }
     }
 
+    private boolean isAttemptingPasswordChange( CliArgs args ) {
+        return args.getCypher().map(  (s) -> s.toUpperCase().startsWith( "ALTER CURRENT USER SET PASSWORD FROM" ) ).orElse( false );
```

Comment:
Isn't this detection a bit weak w.r.t e.g. comments in the beginning of a `.cypher` file or formatting whitespace, etc. ?